public/app/features/correlations/CorrelationsPage.tsx

```diff
@@ -0,0 +1,204 @@
+import { css } from '@emotion/css';
+import { negate } from 'lodash';
+import React, { ComponentProps, memo, useCallback, useMemo, useState } from 'react';
+import { CellProps, SortByFn } from 'react-table';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { Badge, Button, DeleteButton, HorizontalGroup, LoadingPlaceholder, useStyles2, Alert } from '@grafana/ui';
+import { Page } from 'app/core/components/Page/Page';
+import { contextSrv } from 'app/core/core';
+import { useNavModel } from 'app/core/hooks/useNavModel';
+import { AccessControlAction } from 'app/types';
+
+import { AddCorrelationForm } from './Forms/AddCorrelationForm';
+import { EditCorrelationForm } from './Forms/EditCorrelationForm';
+import { EmptyCorrelationsCTA } from './components/EmptyCorrelationsCTA';
+import { Column, Table } from './components/Table';
+import { CorrelationData, useCorrelations } from './useCorrelations';
+
+const sortDatasource: SortByFn<CorrelationData> = (a, b, column) =>
+  a.values[column].name.localeCompare(b.values[column].name);
+
+const isSourceReadOnly = ({ source }: Pick<CorrelationData, 'source'>) => source.readOnly;
+
+const loaderWrapper = css`
+  display: flex;
+  justify-content: center;
+`;
+
+export default function CorrelationsPage() {
+  const navModel = useNavModel('correlations');
+  const [isAdding, setIsAdding] = useState(false);
+  const { correlations, create, remove, update, isLoading, error, reload } = useCorrelations();
+  const canWriteCorrelations = contextSrv.hasPermission(AccessControlAction.DataSourcesWrite);
+
+  const handleAdd = useCallback<ComponentProps<typeof AddCorrelationForm>['onSubmit']>(
+    async (correlation) => {
+      await create(correlation);
+      reload();
```

Comment:
`create` already calls `reload`, isn't?

Reply:
Sorry, commented after you changed it. NVM.

Reply:
this is why https://github.com/grafana/grafana/pull/53821#issuecomment-1220671286 was happening, removed in https://github.com/grafana/grafana/pull/53821/commits/fbf708d3f287fc29323a94d1b4e3c82dda9d0d94